course8/course_en.md

```diff
@@ -0,0 +1,184 @@
+# Modern Programming Ideology: Implementing Queues with Mutable Data Structures
+
+## Overview
+
+In modern programming concepts, a queue is an important data structure that follows the First-In-First-Out (FIFO) principle. This course will show you how to use mutable data structures to implement queues, focusing on two implementation methods: circular queues and singly linked lists, and exploring the concepts of tail call and tail recursion.
+
+## Basic Operations of Queues
+
+A queue supports the following basic operations:
+
+- `make()`: Creates an empty queue.
+- `push(t: Int)`: Adds an integer element to the queue.
+- `pop()`: Removes an element from the queue.
+- `peek()`: Views the front element of the queue.
+- `length()`: Obtains the length of the queue.
+
+## Implementation of Circular Queues
+
+Circular queues implement queues using arrays, which provide a continuous storage space where each position can be modified. Once an array is allocated, its length remains fixed.
+
+### Creating an Array
+
+```moonbit expr
+let a: Array[Int] = Array::make(5, 0)
+```
+
+### Adding Elements
+
+```moonbit expr
+let a: Array[Int] = Array::make(5, 0)
+a[0] = 1
+a[1] = 2
+println(a) // Output: [1, 2, 0, 0, 0]
+```
+
+### Simple Implementation of Circular Queues
+
+```moonbit no-check
```

Comment:
```suggestion
```moonbit
```

Reply:
there is queue inside std so can't pass check